### ISSUE TITLE
Catch pydantic validation errors with one validation type

### ISSUE THREAD

**Issue by ketozhang** [ketozhang/asdf-pydantic#2]
> I wonder if it is possible to modify the AsdfPydanticModel to raise the same [ValidationError](https://github.com/asdf-format/asdf/blob/2521da0226fd5e6c3affc3630105a663c0b74de9/asdf/exceptions.py#L9) as ASDF does? That way libraries reading asdf files will on need to worry about catching one error type.
@WilliamJamieson

**Comment by ketozhang** [comment]
ASDF's error is a direct use of jsonschema's which is not a subclass of a standard type
```py
from jsonschema import ValidationError
```

Pydantic's error is a subclass of TypeError
```py
from pydantic.error_wrappers import ValidationError
```

---

I really would love to use ExceptionGroup (new in Python 3.11) for this because there are situations where I want to catch only pydantic validation error.

**Comment by WilliamJamieson** [comment]
Maybe `asdf` should define a `AsdfValidationError` which inherits from both `TypeError` and `jsonschema.ValidationError`? I think that would be a reasonable improvement on `asdf`'s part.

**Comment by ketozhang** [comment]
Desired result:

```py
from asdf import ValidationError

try:
    af.open(...)
except ValidationError:
    ...
```

After Python 3.11,

```py
from asdf import ValidationError
# Very verbose, could forward this asdf_pydantic
from pydantic.error_wrappers import ValidationError as PydanticValidationError  

try:
    af.open(...)
except* ValidationError:
    ...
except* PydanticValidationError:
   ...
```

**Comment by ketozhang** [comment]
Good idea on inheriting `TypeError`. I'd keep the same name `ValidationError` for compatibilty.

> Conversely, since this is such a new library you could just require Python 3.11 to use it.

I will consider it; I have things in Roman that are 3.10.
